py/desispec/scripts/qsoqn.py

```diff
@@ -209,55 +210,61 @@ def extract_redshift_info_from_RR(filename_redrock, targetid):
-        sel = (z_qn >= zmin) & (z_qn <= zmax)
+        zmin = min(zmin, np.min(redshift_template))
+        zmax = max(zmax, np.max(redshift_template))
 
-        # In case where all the objects have priors outside the redshift template range
-        # Skip the template to avoid any undesired errors
-        if sel.sum() != 0:
-            write_prior_for_RR(targetid[sel], z_qn[sel], z_prior[sel], filename_priors)
+    sel = (z_qn >= zmin) & (z_qn <= zmax)
 
-            # Info: in the case where targetid is -7008279, we cannot use it at first element of the targetid list
-            # otherwise RR required at least one argument for --targetids .. (it is a known problem in python this comes from -)
-            # see for example https://webdevdesigner.com/q/can-t-get-argparse-to-read-quoted-string-with-dashes-in-it-47556/
-            # To figure out with this, we just need to add a space before the -
+    # In case where all the objects have priors outside the redshift template range
+    # Skip the template to avoid any undesired errors
+    if sel.sum() != 0:
+        write_prior_for_RR(targetid[sel], z_qn[sel], z_prior[sel], filename_priors)
 
-            argument_for_rerun_RR = ['--infiles', spectra_name,
-                                     '--templates', template_filename,
-                                     '--targetids', ' ' + ",".join(reversed(targetid[sel].astype(str))),
-                                     '--priors', filename_priors,
-                                     '--details', filename_output_rerun_RR,
-                                     '--outfile', filename_redrock_rerun_RR]
-            # NEW RUN RR
-            log.info(f'Running redrock with priors on selected targets with {template_filename}')
-            log.info('rrdesi '+' '.join(argument_for_rerun_RR))
+        # Info: in the case where targetid is -7008279, we cannot use it at first element of the targetid list
+        # otherwise RR required at least one argument for --targetids .. (it is a known problem in python this comes from -)
+        # see for example https://webdevdesigner.com/q/can-t-get-argparse-to-read-quoted-string-with-dashes-in-it-47556/
+        # To figure out with this, we just need to add a space before the -
 
-            rrdesi(argument_for_rerun_RR, comm=comm)
+        argument_for_rerun_RR  = ['--infiles', spectra_name]
+        argument_for_rerun_RR += ['--templates',] + param_RR['templates_filename']
+        argument_for_rerun_RR += ['--targetids', ' '+",".join(reversed(targetid[sel].astype(str))),
+                                                 # see long comment above about need for preceeding space
+                                 '--priors', filename_priors,
+                                 '--details', filename_output_rerun_RR,
+                                 '--outfile', filename_redrock_rerun_RR]
+        # NEW RUN RR
+        log.info(f'Running redrock with priors on selected targets with {template_filename}')
```

Comment:
Now that `param_RR['templates_filename']` is a list of files, I think this should maybe be `param_RR['templates_filename']`? Otherwise you are only listing the last file, and this could even raise an error if `param_RR['templates_filename']` is an empty list such that `template_filename` is never defined.

Reply:
good point; log message updated